File: PairsofSongsWithTotalDurationsDivisibleby60.py
class Solution(object):
    def numPairsDivisibleBy60(self, time):
        """
        :type time: List[int]
        :rtype: int
        """
        rem = [0] * 60
        res = 0
        for x in time:
            c_rem = x % 60
            if (c_rem == 0):
                res += rem[0]
            else:
                res += rem[60 - c_rem]

            rem[c_rem] += 1

        return res

File: java_version/src/PairsofSongsWithTotalDurationsDivisibleby60.java
public class PairsofSongsWithTotalDurationsDivisibleby60 {
    public int numPairsDivisibleBy60(int[] time) {
        int[] rem = new int[60];
        int count = 0;
        for(int x: time){
            int c_rem = x%60;
            if(c_rem == 0)
                count += rem[0];
            else
                count += rem[60 - c_rem];
            rem[c_rem] += 1;
        }
        return count;
    }
}
